assets/src/apps/authoring/components/PropertyEditor/custom/CompJsonEditor.tsx

```diff
@@ -21,8 +21,9 @@ const CompJsonEditor: React.FC<JsonEditorProps> = (props) => {
   };
   useEffect(() => {
     try {
+      setValue(JSON.stringify(jsonValue, null, 4));
       const jsonVal = JSON.parse(value);
-      if (existingPartIds.indexOf(jsonVal.id) !== -1 && currentPartSelection !== jsonVal.id) {
+      if (existingPartIds.indexOf(jsonValue.id) !== -1 && currentPartSelection !== jsonValue.id) {
```

Comment:
why only using `jsonValue` for *some* of these checks?

Reply:
That is the changed value passed to compJsonEditor.